File: templates/go/team_source/strategy.go
package main

import rand "math/rand/v2"

// ChooseMove returns one legal move using only the wrapper-provided random stream.
func ChooseMove(turn int, myHistory, opponentHistory string, rng *rand.Rand) string {
	moves := [...]string{"R", "P", "S"}
	return moves[rng.IntN(len(moves))]
}

File: templates/typescript/team_source/strategy.ts
export function chooseMove(
  turn: number,
  myHistory: string,
  opponentHistory: string,
  rng: { nextInt(upperExclusive: number): number }
): string {
  const moves = ["R", "P", "S"];
  return moves[rng.nextInt(moves.length)];
}
